src/plugins/content/itemScrolling/itemScrolling.js

```diff
@@ -36,81 +37,112 @@ export default pluginFactory({
      */
     init: function init() {
         const testRunner = this.getTestRunner();
-        const $contentArea = testRunner
-            .getAreaBroker()
-            .getContentArea();
-
-        const gridRowBottomMargin = 12,
-            qtiItemPadding = 30 * 2;
+        const $contentArea = testRunner.getAreaBroker().getContentArea();
 
         testRunner
-            .on('renderitem', function() {
-                adaptItemHeight();
+            .on('renderitem', function () {
+                const isVerticalWritingMode = getIsItemWritingModeVerticalRl();
+                if (isVerticalWritingMode) {
+                    // qti-itemBody is item scroll container in vertical writing mode (but not in horizontal);
+                    // window resize is not enough, because if review-panel is used, it can be closed/opened, and that affects qti-itemBody width.
+                    const $itemBody = $('.qti-itemBody');
+                    this.itemBodyResizeObserver = new ResizeObserver(() => requestAnimationFrame(adaptItemBlockSize));
```

Comment:
Does the nested `requestAnimationFrame()` limit the calls to `getBoundingClientRect()` and prevent any performance impact? (Normally `gBCR` should not be called in a `resize` handler, but I vaguely remember there is some trick with `rAF` to mitigate it.)

Reply:
Without `requestAnimationFrame` there's a console error about ResizeObserver loop. 
Callback is called only once for me.

In _vertical-writing_, `.qti-itemBody` is a scroll container and has fixed/heigth, so it shouldn't be influenced by its content.
In _horizontal-writing_, `.qti-itemBody` fits its content, so you can't really use this ResizeObserver to measure space available for shared-stimulus --> this observer is not attached for horizontal-writing.